### DISCUSSION TITLE
“Disentangling failed: 'Multiple isolated embplant_pt components detected! Broken or contamination?” AND the assemble length range different from reference

### DISCUSSION THREAD

**Discussion by GaoXY99** [Kinggerm/GetOrganelle#324] [Q&A]
Hi, I have some problems during chloroplast genome assembling and need your help. I ran several times, upload the following two commands, log, and the results.  Thank you very much for your attentiion.

COMMAND1:
get_organelle_from_reads.py -1 seq/PYA1/PYA1_clean_R1.fastq.gz -2 seq/PYA1/PYA1_clean_R2.fastq.gz -o temp/round50_k12_seedetc/PYA1_new -t 24 -F embplant_pt -R 50 -k 21,29,39,45,59,65,79,85,99,105,119,141 
![parameter1](https://github.com/Kinggerm/GetOrganelle/assets/22140168/b8c3534d-6d71-44b6-a613-e02b8d34d50c)
[parameter1_get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/13959624/parameter1_get_org.log.txt)

COMMAND2:
get_organelle_from_reads.py -1 seq/PYA1/PYA1_clean_R1.fastq.gz -2 seq/PYA1/PYA1_clean_R2.fastq.gz -o temp/round50_k12_seedetc/PYA1_new -t 24 -F embplant_pt -R 50 -k 21,29,39,45,59,65,79,85,99,105,119,141 -w 85 --max-reads inf --reduce-reads-for-coverage inf
![parameter2](https://github.com/Kinggerm/GetOrganelle/assets/22140168/4efd4ef8-0230-4bab-8948-0f3db3785d4e)
[parameter2_get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/13959629/parameter2_get_org.log.txt)

1. Disentangling fail reported in these two log file
Disentangling failed: 'Multiple isolated embplant_pt components detected! Broken or contamination?'  
Disentangling failed: 'No new connections.'
How to deal with it?

2. assembled sequence length of quadripartite is obviously different from the reference
eg: assembled SSC sequence length is 13kb, the SSC of reference is 18kb
      assembled IR (32 kb) is longer than the reference(27kb).

3. It seemed that "-w 85 --max-reads inf --reduce-reads-for-coverage inf" improve the depth, however, it has not reduced the contig number.

**Comment by JianjunJin** [comment]
Thank you for your query. Based on your runs, it appears that the organelle sufficient graph has already been achieved, so adjusting parameters may not be necessary (refer to our [FAQ](https://github.com/Kinggerm/GetOrganelle/wiki/FAQ#what-should-i-do-with-incomplete-resultbroken-assembly-graph) for more details).

The primary challenge is the disentanglement of graphs with extremely high depth contigs. We have a new branch addressing this issue, although it's not yet officially released on conda. You can check it out [here - the intermediate_graph branch](https://github.com/Kinggerm/GetOrganelle/tree/intermediate_graph). Since it's not available on bioconda, please install using [Option 2 or 3](https://github.com/Kinggerm/GetOrganelle/wiki/Installation#option-2-using-the-setuppy). Don't hesitate to contact me for any assistance.

**Comment by smallfishcui** [comment]
I encountered the same error while using the assembly done by flye. I adjusted the parameter --min-depth 100 --max-depth 3000 and it finally completed the analyses. I changed the parameters for multiple times...it doesn't work if the range is smaller or larger than these.
best,
Cui

**Reply by JianjunJin** [reply]
Thanks for reachingout. However, it will be informative only when you provide the log file and the graph file.
